src/main/java/com/kodilla/ecommercee/service/CartService.java

```diff
@@ -0,0 +1,82 @@
+package com.kodilla.ecommercee.service;
+
+import com.kodilla.ecommercee.controller.exception.CartNotFoundException;
+import com.kodilla.ecommercee.controller.exception.ItemNotFoundException;
+import com.kodilla.ecommercee.controller.exception.ProductNotFoundException;
+import com.kodilla.ecommercee.controller.exception.UserNotFoundException;
+import com.kodilla.ecommercee.domain.carts.Cart;
+import com.kodilla.ecommercee.domain.orders.Item;
+import com.kodilla.ecommercee.domain.orders.Order;
+import com.kodilla.ecommercee.domain.products.Product;
+import com.kodilla.ecommercee.repository.CartRepository;
+import com.kodilla.ecommercee.repository.OrderRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.transaction.Transactional;
+import java.time.LocalDate;
+import java.util.List;
+
+@Service
+@Transactional
+public class CartService {
+    private final CartRepository cartRepository;
+    private final ProductService productService;
+    private final ItemService itemService;
+    private final OrderRepository orderRepository;
+
+    @Autowired
+    public CartService(CartRepository cartRepository, ProductService productService,
+                       ItemService itemService, OrderRepository orderRepository) {
+        this.cartRepository = cartRepository;
+        this.productService = productService;
+        this.itemService = itemService;
+        this.orderRepository = orderRepository;
+    }
+
+    public List<Item> getAllProductInCart(long cartId) throws CartNotFoundException {
+        List<Product> products = productService.getAllProducts();
+        List<Item> items = cartRepository.findById(cartId).orElseThrow(CartNotFoundException::new).getItems();
+        for (int i = 0; i < items.size(); i++) {
+            Item item = items.get(i);
+            Long productIdOfItem = item.getProductId();
+            for (int j = 0; j < products.size(); j++) {
+                Long productIdOfProduct = products.get(j).getProductId();
+                if (productIdOfItem.equals(productIdOfProduct)) {
+                    Item itemToChangePrice = itemService.getItem(item.getId());
+                    itemToChangePrice.setPrice(productService.getProduct(productIdOfProduct).getPrice());
+                    itemService.saveItem(itemToChangePrice);
+                }
+            }
+        }
+        return items;
+    }
```

Comment:
pętla powoduję to, że jak zmieni się cene produktu, to w koszyku też sie zmieni cena. 
Planuję potem dodać jakiś Observer żeby pętla wykonywała się tylko wtedy keidy zostanie zaktualizowany Produkt, tak zeby zakażdym razem nie była wywoływana podczas sprawdzenia koszyka